code/modules/reagents/chemistry/reagents/medicine.dm

```diff
@@ -226,16 +248,28 @@
 	update_flags |= M.adjustFireLoss(-2*REAGENTS_EFFECT_MULTIPLIER, FALSE)
 	return ..() | update_flags
 
+/datum/reagent/medicine/silver_sulfadiazine/heal_external_limb(obj/item/organ/external/organ, volume)
+	organ.heal_damage(0, volume, updating_health=FALSE)
+
 /datum/reagent/medicine/silver_sulfadiazine/reaction_mob(mob/living/M, method=REAGENT_TOUCH, volume, show_message = 1)
 	if(iscarbon(M))
+		if(!ishuman(M) && method == REAGENT_SPLASH) // we dont use specified logic for humans if target is not human
+			method = REAGENT_TOUCH
+
 		if(method == REAGENT_TOUCH)
 			M.adjustFireLoss(-volume)
 			if(show_message)
 				to_chat(M, "<span class='notice'>The silver sulfadiazine soothes your burns.</span>")
+
 		if(method == REAGENT_INGEST)
 			M.adjustToxLoss(0.5*volume)
 			if(show_message)
 				to_chat(M, "<span class='warning'>You feel sick...</span>")
+
+		if(ishuman(M) && method == REAGENT_SPLASH)
+			splash_human(M, volume)
+			if(show_message)
+				to_chat(M, "<span class='notice'>The silver sulfadiazine soothes your burns.</span>")
 	..()
```

Comment:
This is code spaghetti, you shouldn't and don't need to change the reaction method just for this; this can also be simplified and deduplicated (same applies to Styptic Powder/Silver Sulfadiazine)
```suggestion
		if(method == REAGENT_TOUCH || method == REAGENT_SPLASH)
			if(ishuman(M))
				splash_human(M, volume)
			else
				M.adjustFireLoss(-volume)
			if(show_message)
				to_chat(M, "<span class='notice'>The silver sulfadiazine soothes your burns.</span>")
		else if(method == REAGENT_INGEST)
			M.adjustToxLoss(0.5*volume)
			if(show_message)
				to_chat(M, "<span class='warning'>You feel sick...</span>")
	..()
```

Reply:
this is not the same logic. When REAGENT_TOUCH applied to human, it shouldnt call splash_human
i will look into it later, if i get any ideas to make it look better, i will update the PR

Reply:
> When REAGENT_TOUCH applied to human, it shouldnt call splash_human
right, add `method == REAGENT_SPLASH` to the `if(ishuman(M))` check then